components/overflow-group/overflow-group.js

```diff
@@ -0,0 +1,495 @@
+			if (this.maxToShow >= 0 && showing.count >= this.maxToShow) {
+				isForcedOverflowing = true;
+				itemLayout.isChomped = true;
+				itemLayout.trigger = 'force-hide';
+				continue;
+			}
+
+			// chomp or unchomp based on space available, and we've already handled min/max above
+			if (!isSoftOverflowing && showing.width + itemLayout.width < this._availableWidth) {
+				showing.width += itemLayout.width;
+				showing.count += 1;
+				itemLayout.isChomped = false;
+				itemLayout.trigger = 'soft-show';
+
+			} else {
+				isSoftOverflowing = true;
+				itemLayout.isChomped = true;
+				itemLayout.trigger = 'soft-hide';
+
+			}
+
+		}
+		// if there is at least one showing and no more to be hidden, enable collapsing more button to [...]
+		this._overflowMenuHidden = items.length === showing.count;
+		if (!this._overflowMenuHidden && (isSoftOverflowing || isForcedOverflowing)) {
+			for (let j = this._layoutItems.length; j--;) {
+				if (showing.width + this._overflowMenuWidth < this._availableWidth) {
+					break;
+				}
+				const itemLayoutOverflowing = this._layoutItems[j];
+				if (itemLayoutOverflowing.trigger !== 'soft-show') {
+					continue;
+				}
+				showing.width -= itemLayoutOverflowing.width;
+				showing.count -= 1;
+				isSoftOverflowing = true;
+				itemLayoutOverflowing.trigger = 'soft-hide';
+				itemLayoutOverflowing.isChomped = true;
+			}
+		}
+		const overflowDropdownOverflowing = (showing.width + this._overflowMenuWidth >= this._availableWidth);
+		const swapToMiniButton = overflowDropdownOverflowing && !this._overflowMenuHidden;
+
+		this._mini = this.openerType === OPENER_TYPE.ICON || swapToMiniButton;
+		this._chompIndex = this._overflowMenuHidden ? null : showing.count;
+
+		this.dispatchEvent(new CustomEvent('d2l-overflow-group-updated', { bubbles: true, composed: true }));
+	}
+	_convertToDropdownItem(node) {
+		const tagName = node.tagName.toLowerCase();
+		let menuItem;
+		if (tagName === 'd2l-button' || tagName === 'd2l-button-subtle' || tagName === 'button' || tagName === 'd2l-button-icon') {
+			menuItem = this._createMenuItem(node);
+		} else if (tagName === 'd2l-link' || tagName === 'a') {
+			menuItem = this._createMenuItemLink(node);
+		} else if (node.getAttribute('role') === 'separator') {
+			menuItem = this._createMenuItemSeparator();
+		} else if (tagName === 'd2l-menu' || tagName === 'd2l-dropdown' || tagName === 'd2l-dropdown-button' || tagName === 'd2l-dropdown-button-subtle') {
+			menuItem = this._createMenuItemMenu(node);
+		} else if (tagName === 'd2l-menu-item') {
+			// if the menu item has children treat it as a menu item menu
+			if (node.children.length > 0) {
+				menuItem = this._createMenuItemMenu(node);
+			} else {
+				menuItem = this._createMenuItem(node);
+			}
+		} else if (node.classList.contains('d2l-overflow-group-custom-item')) {
+			menuItem = this._createMenuItem(node);
+		} else {
+			return;
+		}
+		return menuItem;
+	}
+	_createMenuItem(node) {
```

Comment:
It looks like most of these `_create*` methods are just helpers and don't actually reference `this`. So I think they can just be pulled outside of the class to the top of the file as functions.

It doesn't make a huge difference, but that way they'll be treated as static and shared amongst all the instances of this class, vs. having copies of them for each one.